src/seahawk/seahawk_deck/set_remote_params.py

```diff
@@ -0,0 +1,119 @@
+        """
+        # Create service name. Services to set parameters must be named as '/other_node_name/set_parameters'
+        srv_name = "/" + other_node_name + "/set_parameters"
+
+        # Set up client to remotely set parameters of another node using a service
+        self.__cli = this_node.create_client(SetParameters, srv_name)
+
+        # Try to connect to the service (wait one second between attempts)
+        while not self.__cli.wait_for_service(timeout_sec=1.0):
+            this_node.get_logger().info(f"{srv_name} service not available, waiting again...")
+        
+        # Create a set parameter request object
+        self.__req = SetParameters.Request()
+
+        # List to store parameters before sending them
+        self.__param_list = []
+
+    def update_params(self, param_name: str, param_value: Any):
+        """
+        Appends a single new parameter given its name and new value to the list of parameters to set
+
+        Args: 
+            param_name: Name of the parameter
+            param_value: Value of the parameter (note must of a be a valid parameter type)
+        """
+        self.__param_list.append(Parameter(name=param_name, value=param_value).to_parameter_msg())
+	
+    def send_params(self) -> (Any | None):
+        """
+        Sends updated parameter list to the service
+
+        Returns:
+            The result set by the task, None if no result was set
+        """
+        self.__req.parameters = self.__param_list   # Create updated param list
+        future = self.__cli.call_async(self.__req)  # Send param to service
+        self.__param_list.clear()                   # Reset list
+        return future.result()                      # Return if the parameter is set correctly
+    
+
+class SrvRemoteParams():
+    """
+    Service node. Abstracts requesting setting node's parameters from another node via service to a class.
+    """
+    def __init__(self, this_node: Node):
+        """
+        Set up request to setting thi node's parameters from another node
+    
+        Args:
+            this_node: Name of the node whose parameters should be set. Service node
+        """
+        self.__this_node = this_node
+        # Call parameter callback
+        self.__this_node.add_on_set_parameters_callback(self.__set_params)
+    
+    def __set_params(self, params: list[Parameter]) -> SetParametersResult:
+        """
+        Callback for parameter update. Updates all local parameters to have values set by the service
+
+        Args:
+            params: List of updated parameters
+
+        Returns:
+            SetParametersResult() which lets ROS2 know if the parameters were set correctly or not
+        """
+        try:
+            # Try to set parameters to updated values
+            self.__this_node.set_parameters(params)
+        except:
+            return SetParametersResult(successful=False)
+        return SetParametersResult(successful=True)
```

Comment:
This is effectively re-implementing what ROS is already doing, setting the parameters. This leads to unpredictable behavior, likely recursive.

https://docs.ros.org/en/rolling/Concepts/Basic/About-Parameters.html#parameter-callbacks
> A ROS 2 node can register three different types of callbacks to be informed when changes are happening to parameters. All three of the callbacks are optional.
> ...
> The second is known as a “set parameter” callback, and can be set by calling `add_on_set_parameters_callback` from the node API. The callback is passed a list of immutable `Parameter` objects, and returns an `rcl_interfaces/msg/SetParametersResult`. The main purpose of this callback is to give the user the ability to inspect the upcoming change to the parameter and explicitly reject the change.
> ...

here's an example of what `add_on_set_parameters_callback` is supposed to be doing: https://github.com/ros2/demos/blob/rolling/demo_nodes_cpp/src/parameters/set_parameters_callback.cpp#L70

Reply:
Nice catch, I removed the entire `SrvRemoteParams` class and it still works